### PR TITLE
Cache network settings

### LINKED ISSUE #247: rest gw doesn't take into account updated network settings
https://rest.fs.neo.org/HXSaMJXk2g8C14ht8HSi7BBaiYZ1HeWh2xnWPGQCg4H6/518-1726084393/index.html#suites/820837d46c5808d6efecf59b97ab3206/ad221a173e92ac0d/

1. Set HomomorphicHashingDisabled=true
2. Create new container via neofs-cli - success
3. Try to create new container via rest-gw - failed:

ir logs:
```
2024-09-11T18:38:54.035Z	error	container/process_container.go:46	put container check failed	{"error": "incorrect homomorphic hashing setting: network setting: true, container setting: false"}
```

Either this setting should be available via rest gw api to set, or it should be taken from the network.

### PR DESCRIPTION
**Pull request by smallhive**
Closes #247.

### CODE REVIEW

File: internal/cache/networkinfo.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	"github.com/nspcc-dev/neofs-sdk-go/client"
+	"github.com/nspcc-dev/neofs-sdk-go/netmap"
+	"github.com/nspcc-dev/neofs-sdk-go/pool"
+)
+
+type (
+	// NetworkInfo is cache wrapper for the network info.
+	NetworkInfo struct {
+		p     *pool.Pool
+		cache *expirable.LRU[string, netmap.NetworkInfo]
```
Comment (roman-khimov) [inline internal/cache/networkinfo.go:18]: But it's a single value that we ever store here, why LRU?

**Review by roman-khimov — COMMENTED** [review]
Do we have any other places where network info is used? Can they reuse this cache?

File: internal/cache/networkinfo.go (hunk 1 above)
Comment (smallhive) [reply internal/cache/networkinfo.go:18]: I used cache already in the project, and decided not to add another lib. You have a better option, I don't mind to replace it

Comment (roman-khimov) [reply internal/cache/networkinfo.go:18]: Just drop LRU? Have an instance of `netmap.NetworkInfo`, have some timestamp associated with it.

Comment (smallhive) [reply internal/cache/networkinfo.go]: It is an option as well, removed

File: cmd/neofs-rest-gw/config.go [hunk 2]
```diff
@@ -596,6 +596,7 @@ func newNeofsAPI(ctx context.Context, logger *zap.Logger, v *viper.Viper) (*hand
 	}
 
 	apiPrm.DefaultTimestamp = v.GetBool(cfgPoolDefaultTimestamp)
+	apiPrm.EpochUpdateInterval = time.Duration(int64(ni.EpochDuration())/2*ni.MsPerBlock()) * time.Millisecond
```
Comment (roman-khimov) [inline cmd/neofs-rest-gw/config.go:599]: It can change, btw.

Comment (roman-khimov) [reply cmd/neofs-rest-gw/config.go:599]: You don't need this parameter now, it all can be handled by `NetworkInfo`.

File: handlers/containers.go [hunk 3]
```diff
@@ -406,6 +406,15 @@ func createContainer(ctx context.Context, p *pool.Pool, stoken session.Container
 	cnr.SetBasicACL(basicACL)
 	cnr.SetOwner(stoken.Issuer())
 
+	ni, err := networkInfoGetter.NetworkInfo(ctx)
```
Comment (roman-khimov) [inline handlers/containers.go:409]: And this is a separate fix.

**Review by roman-khimov — COMMENTED** [review]
Please also update `getEpochDurations` and `getCurrentEpoch`.

File: internal/cache/networkinfo.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+	"github.com/nspcc-dev/neofs-sdk-go/pool"
+)
+
+type (
+	// NetworkInfo is cache wrapper for the network info.
+	NetworkInfo struct {
+		p   *pool.Pool
+		ttl time.Duration
+
+		mu         *sync.RWMutex
```
[the remark this answers is not in the document]
Comment (smallhive) [reply internal/cache/networkinfo.go]: Replaced with Mutex

**Review by roman-khimov — APPROVED** [review]

**Comment by roman-khimov** [comment]
In general it's OK, but now some tests fail here: https://rest.fs.neo.org/7ZVSQHtioVRiDT2btN9xEmQhczXV9vryCEAHUawfD8Sc/141-1729604693/index.html#suites/21ade783424f8b7f50534d1bb781c71b/2824020f8d392ae/

@evgeniiz321, is there any forced epoch tick there? Can they be fixed to handle cached epochs?

**Pull request merged**